src/ray/common/status.h

```diff
@@ -49,7 +49,7 @@ class error_code;
 // Return the given status if it is not OK.
 #define RAY_RETURN_NOT_OK(s)           \
   do {                                 \
-    ::ray::Status _s = (s);            \
```

Comment:
The copy happens on critical path;
If passed-in `s` is a expression, it would be evaluated first, constant reference would do lifecycle extension.